reframe/frontend/statistics.py

```diff
@@ -70,46 +74,103 @@ def retry_report(self):
 
         return '\n'.join(report)
 
+    @property
+    def report_dict(self):
+        _report = []
+        current_run = rt.runtime().current_run
+        for tl, tr in zip(self._tasks, range(current_run + 1)):
+            for t in tl:
+                check = t.check
+                partition = check.current_partition
+                partname = partition.fullname if partition else None
+                environ_name = (check.current_environ.name
+                                if check.current_environ else None)
+                report = {
+                    'name': check.name,
+                    'description': check.descr,
+                    'system': partname,
+                    'environment': environ_name,
+                    'tags': list(check.tags),
+                    'maintainers': check.maintainers,
+                    'scheduler': None,
+                    'jobid': None,
+                    'nodelist': [],
+                    'job_stdout': None,
+                    'job_stderr': None,
+                    'build_stdout': None,
+                    'build_stderr': None,
+                    'failing_reason': None,
+                    'failing_phase': None,
+                    'outputdir': None,
+                    'stagedir': None
+                }
+                if check.job:
+                    report['scheduler'] = check.job.scheduler.registered_name
+                    report['jobid'] = check.job.jobid
+                    report['nodelist'] = (check.job.nodelist
+                                          if check.job.nodelist
+                                          else [])
+                    report['job_stdout'] = check.job.stdout
+                    report['job_stderr'] = check.job.stderr
+
+                if check._build_job:
+                    report['build_stdout'] = check._build_job.stdout
+                    report['build_stderr'] = check._build_job.stderr
+
+                if t.failed:
+                    report['result'] = 'fail'
+                    if t.exc_info is not None:
+                        from reframe.core.exceptions import format_exception
+
+                        report['failing_reason'] = format_exception(
+                            *t.exc_info)
+                        report['failing_phase'] = t.failed_stage
+                        report['stagedir'] = check.stagedir
+                else:
+                    report['result'] = 'success'
+                    report['outputdir'] = check.outputdir
+
+                report['try'] = tr
```

Comment:
Replace with `run_no` as suggested below:

```suggestion
                report['run_no'] = run_no
```